pkg/model/context.go

```diff
@@ -42,23 +46,42 @@ type KopsModelContext struct {
 
 // Will attempt to calculate a meaningful name for an ELB given a prefix
 // Will never return a string longer than 32 chars
-func (m *KopsModelContext) GetELBName32(prefix string) (string, error) {
-	var returnString string
+// Note this is _not_ the primary identifier for the ELB - we use the Name tag for that.
+func (m *KopsModelContext) GetELBName32(prefix string) string {
 	c := m.Cluster.ObjectMeta.Name
-	s := strings.Split(c, ".")
 
-	// TODO: We used to have this...
-	//master-{{ replace .ClusterName "." "-" }}
-	// TODO: strings.Split cannot return empty
-	if len(s) > 0 {
-		returnString = fmt.Sprintf("%s-%s", prefix, s[0])
-	} else {
-		returnString = fmt.Sprintf("%s-%s", prefix, c)
+	if UseLegacyELBName.Enabled() {
+		tokens := strings.Split(c, ".")
+		s := fmt.Sprintf("%s-%s", prefix, tokens[0])
+		if len(s) > 32 {
+			s = s[:32]
+		}
+		glog.Infof("UseLegacyELBName feature-flag is set; built legacy name %q", s)
+		return s
 	}
-	if len(returnString) > 32 {
-		returnString = returnString[:32]
+
+	// The LoadBalancerName is exposed publicly as the DNS name for the load balancer.
+	// So this will likely become visible in a CNAME record - this is potentially some
+	// information leakage.
+	// But... if a user can see the CNAME record, they can see the actual record also,
+	// which will be the full cluster name.
+	s := prefix + "-" + strings.Replace(c, ".", "-", -1)
+	if len(s) > 32 {
```

Comment:
Interesting and a good point... I'm inclined to agree, but I'll think for a bit on it.

I think we will now end up with more of the cluster name in the ELB name...